### PR TITLE
Jetpack Checkout: submit website URL to WPCOM endpoint

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

* Make the `Your website address:` input box located in the post-purchase Thank You page submit the URL to WPCOM.

#### Testing instructions

* Patch your sandbox with D63168-code.
* Patch your sandbox with D63915-code.
* Sandbox `public-api.wordpress.com`.
* Complete the site-less checkout flow.
* On the Thank You page, enter a URL.
* Verify that the URL was appended to the ticket as a private note.

Related to 1200479326344990-as-1200571786740545

### CODE REVIEW

**Review by elliottprogrammer — COMMENTED** [review]
I'm not sure if this was ready for review yet, but I tested this already and it works great! Code looks great! Nice work @rcanepa ! 👍
Let me know when it's ready for review and I can test it again, if you want. 🙂

**Comment by rcanepa** [comment]
> I'm not sure if this was ready for review yet, but I tested this already and it works great! Code looks great! Nice work @rcanepa ! 👍
> Let me know when it's ready for review and I can test it again, if you want. 🙂

Actually, it is ready, so thank you for reviewing it. I was waiting for the backend patch to be shipped before I officially open this for review (to avoid asking people to do a complicated setup on their sandboxes).

**Comment by rcanepa** [comment]
@elliottprogrammer, I officially opened this PR for review. I would appreciate a second round of testing. Let's make sure that this time the receipt ID is added to the ticket subject and that everything still works as expected. For the record, the subject of a ticket should follow the pattern `Jetpack Site-less Purchase [ReceiptID:receiptId]`.

File: client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts [hunk 1]
```diff
@@ -141,7 +141,12 @@ export default function getThankYouPageUrl( {
 		// extract a product from the cart, in siteless checkout there should only be one
 		const productSlug = cart?.products[ 0 ]?.product_slug;
 
-		return `/checkout/jetpack/thank-you/no-site/${ productSlug ?? 'no_product' }`;
+		const thankYouUrlSiteLess = `/checkout/jetpack/thank-you/no-site/${
+			productSlug ?? 'no_product'
+		}`;
+		return pendingOrReceiptId
+			? `${ thankYouUrlSiteLess }?receiptId=${ pendingOrReceiptId }`
```
Comment (sirbrillig) [inline client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts:151]: I'm sure there's a unit test for this URL... could you update it to cover the case where it gets this query param?

Comment (rcanepa) [reply client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts:151]: Sorry for missing the obviously broken test. I just pushed a commit that makes the logic more robust and adds unit tests to cover it.

Comment (sirbrillig) [reply client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts:151]: 🎉  thank you!

**Review by elliottprogrammer — APPROVED** [review]
Tested again and it works great! The ticket is updated with a note containing the submitted site url and the ticket is set to 'Urgent'.  The receiptId is added to the ticket subject. I see no issues, code looks great! All tests are passing.
LGTM! 👍

**Pull request merged**